File: typescript/src/ui/overlays/next-level.ts
import Colors from '../colors';
import Fonts from '../fonts';
import Overlay from './overlay';

export default class NextLevel extends Overlay {
    private pastMillis: number = 0;
    private displayNextLevel = true;

    draw(level: number) {
        this.p5.push();

        this.p5.fill(Colors.EDGE);
        this.p5.textAlign(this.p5.CENTER);
        this.p5.textSize(60);

        this.p5.text(
            'LEVEL ' + level + ' COMPLETED',
            this.centerX,
            this.centerY
        );

        const now = this.p5.millis();

        if (now - this.pastMillis > 1000) {
            this.displayNextLevel = !this.displayNextLevel;
            this.pastMillis = now;
        }

        if (this.displayNextLevel) {
            this.p5.textFont(Fonts.fontLight);
            this.p5.textSize(25);

            this.p5.text(
                'PRESS "S" TO GO TO THE NEXT LEVEL',
                this.centerX,
                this.centerY + 50
            );
        }

        this.p5.pop();
    }
}

File: rust/src/overlays/next_level.rs
use web_sys::CanvasRenderingContext2d;

use crate::{sprites::sprite::CanvasDimension, utils::javascript::millis};

pub struct NextLevel {
    past_millis: f64,
    display_press_s: bool,
    center_x: f64,
    center_y: f64,
}

impl NextLevel {
    pub fn new(canvas: CanvasDimension) -> NextLevel {
        NextLevel {
            past_millis: 0.0,
            display_press_s: false,
            center_x: canvas.width / 2.0,
            center_y: canvas.height / 2.0,
        }
    }
}

impl NextLevel {
    pub fn update(&mut self) {
        let now = millis();

        if now - self.past_millis > 1000.0 {
            self.display_press_s = !self.display_press_s;
            self.past_millis = now;
        }
    }

    pub fn draw(&self, level: u32, canvas: CanvasRenderingContext2d) {
        canvas.save();

        canvas.set_text_align("center");
        canvas.set_font("100 60px 'Exo 2'");
        let text = format!("LEVEL {} COMPLETED", level);
        let _result = canvas.fill_text(&text, self.center_x, self.center_y);

        if self.display_press_s {
            canvas.set_font("100 25px 'Exo 2'");

            let _result = canvas.fill_text(
                "PRESS \"S\" TO GO TO THE NEXT LEVEL",
                self.center_x,
                self.center_y + 50.0,
            );
        }

        canvas.restore();
    }
}
